src/library/simplified/extension/checkout-date.js

```diff
@@ -5,6 +5,10 @@ if (utag.isHSR() && b['entity.hotels.search.hotelParameters.checkOutDate'])
 {
     b['checkOutDate'] = b['entity.hotels.search.hotelParameters.checkOutDate'];
 }
+else if (utag.isRailSearchResults() && b['entity.railSearch.searchParameters.journeyLegs.1']!= undefined )
+{
+    b["checkOutDate"] = b['entity.railSearch.searchParameters.journeyLegs.1'].isoFormatDate;
```

Comment:
I think I need to correct somemore DataPopulator work in rails-web. I didnt consider ARRIVE By option when correcting DataPopulator. I tried like this and got everything correctly populator. I need to consider the scenario that you have mentioned.

![image](https://cloud.githubusercontent.com/assets/21168911/24899883/90245a52-1ebf-11e7-99d0-f77073845426.png)